operator/pkg/awsprovider/launchtemplate/reconciler.go

```diff
@@ -72,10 +72,35 @@ func (c *Controller) Reconcile(ctx context.Context, dataplane *v1alpha1.DataPlan
 	return nil
 }
 
+func (c *Controller) CreateResource(
```

Comment:
> Why not add these fields to the DataplaneSpec and add the is check for these fields in Create method. If a user provides any of these values use it else try to discover the way we are trying to discover these values today?

Because `securityGroupID` & `clusterEndpoint` are not specifications. They are resources created. `amiID` falls into the category of specification, instaceProfile is a string with a fixed format, but the formats are different in `kubernetes-iteration-toolkit` and `EKSTestClusterCreationCommon`